pipeline_steps/common.groovy

```diff
@@ -684,4 +684,102 @@ void internal_slave(body){
   use_node("CentOS", body)
 }
 
+// Build an array suitable for passing to withCredentials
+// from a space or comma separated list of credential IDs.
+@NonCPS
+List build_creds_array(String list_of_cred_ids){
+    print("Building credentials array from the following list of IDs: ${list_of_cred_ids}")
+    Map creds_bundles = [
+      "cloud_creds": ['dev_pubcloud_username',
+                      'dev_pubcloud_api_key',
+                      'dev_pubcloud_tenant_id'],
+      "rpc_repo": ['RPC_REPO_IP',
```

Comment:
This is inconsistent, cloud_creds is using the id, while rpc_repo creds are using the variable name. I think using the variable would be better for users, will change all to that.

Reply:
On second thoughts, using variable names instead of IPs may lead to problems with credentials that provide multiple variables. Also All the above are in fact IDs, its just inconsistent capitalisation that makes it look like one set are IDs and the other are variables.